internal/catnames/catnames.go

```diff
@@ -0,0 +1,23 @@
+package catnames
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+func NewGenerator(pick func(int) int) func() string {
+	if pick == nil {
+		pick = rand.New(rand.NewSource(time.Now().UnixNano())).Intn
+	}
+
+	return func() string {
+		return fmt.Sprintf(
+			"%s the %s",
```

Comment:
Better english: reverse the order and drop the "the". That's how all the other animal/adjective generators work too.

Reply:
done